src/app/components/hacker-data/view-hacker-data-dialog/view-hacker-data-dialog.ts

```diff
@@ -0,0 +1,310 @@
+    if (!this.dietaryResSelection) { this.hacker.dietary_restriction = null; }
+    if (this.hacker.allergies === 'none' || this.hacker.allergies === 'None') {
+      this.hacker.allergies = 'none';
+      this.user_data.allergies = 'none';
+    };
+    const changedRegistration = {
+      firstName: this.hacker.firstname || this.user_data.firstname,
+      lastName: this.hacker.lastname || this.user_data.lastname,
+      email: this.hacker.email || this.user_data.email,
+      gender: this.hacker.gender || this.user_data.gender,
+      university: this.schoolFormControl.value ? this.schoolFormControl.value : this.user_data.university,
+      academicYear: this.hacker.academic_year || this.user_data.academic_year,
+      major: this.majorFormControl.value ? this.majorFormControl.value : this.user_data.major,
+      shirtSize: this.hacker.shirt_size || this.user_data.shirt_size,
+      dietaryRestriction: this.hacker.dietary_restriction || this.user_data.dietary_restriction,
+      allergies: this.hacker.allergies || this.user_data.allergies,
+      travelReimbursement: this.travelReimToggle.toString() || this.user_data.travel_reimbursement.toString(),
+      firstHackathon: this.firstHackToggle.toString() || this.user_data.first_hackathon.toString(),
+      phone: this.hacker.phone || this.user_data.phone,
+      ethnicity: this.hacker.race || this.user_data.race,
+      codingExperience: this.hacker.coding_experience || this.user_data.coding_experience,
+      eighteenBeforeEvent: this.eighteenBeforeToggle.toString() || this.user_data.eighteenBeforeEvent.toString(),
+      mlhcoc: this.mlhCocToggle.toString() || this.user_data.mlh_coc.toString(),
+      mlhdcp: this.mlhDcpToggle.toString() || this.user_data.mlh_dcp.toString(),
+      referral: this.referralFormControl.value ? this.referralFormControl.value : this.user_data.referral,
+      projectDesc: this.hacker.project || this.user_data.project,
+      expectations: this.hacker.expectations || this.user_data.expectations,
+      veteran: this.veteranToggle.toString() || this.user_data.veteran,
+      uid: this.hacker.uid || this.user_data.uid,
+      submitted: this.hacker.submitted || this.user_data.submitted,
+      hackathon: this.hacker.hackathon || this.user_data.hackathon,
+    }
+    this.dialogRef.close(changedRegistration);
+  }
+
+  /**
+   * Resets the edit form to clear any changes
+   */
+  onResetHackerDataClick() {
+    this.hacker = Object.assign({}, this.user_data);
+    this.schoolFormControl.reset();
+    this.majorFormControl.reset();
+    this.setupToggles();
+  }
+
+  /**
+   * Simple function to convert a single character t-shirt size into a string
+   *
+   * @param: user_size  string representing the size of a t-shirt
+   * @return: String representing the size of a t-shirt
+   */
+  getSizeText(user_size: string) {
+    if (user_size != null) {
```

Comment:
```suggestion
    if (user_size !== null) {
```
A better way is to check the truthy value `if (user_size)` instead, since it could be undefined as well